Realm/ObjectStore/shared_realm.hpp

```diff
@@ -158,6 +158,8 @@ namespace realm {
 
         // FIXME private
         Group *read_group();
+        
+        SharedGroup& shared_group();
```

Comment:
I guess the object store `realm` could have methods to generate the intermediate handover object and another to accept this intermediate object and return the actual handed over object.